firedrake/slate/slate.py

```diff
@@ -444,6 +468,84 @@ def _key(self):
         return (type(self), self._function)
 
 
+class BlockAssembledVector(AssembledVector):
+    """This class is a symbolic representation of an assembled
+    vector of data contained in a set of :class:`firedrake.Function`s
+    defined on pieces of a split mixed function space.
+
+    :arg functions: A tuple of firedrake functions.
+    """
+    def __new__(cls, function, split_functions, indices):
+        if isinstance(split_functions, tuple) \
+           and all(isinstance(f, Coefficient) for f in split_functions):
+            self = TensorBase.__new__(cls)
+            self._function = split_functions
+            self._indices = indices
+            self._original_function = function
+            return self
+        else:
+            raise TypeError("Expecting a tuple of Coefficients or AssembledVectors (not a %r)" %
```

Comment:
I think the check requires coefficients, unless an `AssembledVector` is-a `Coefficient`?